geopandas/geodataframe.py

```diff
@@ -191,6 +191,11 @@ def __init__(self, data=None, *args, geometry=None, crs=None, **kwargs):
             ):
                 raise ValueError(crs_mismatch_error)
 
+            if hasattr(geometry, "name") and geometry.name != "geometry":
```

Comment:
Shall we maybe warn here? I think that would've cleared up my confusion reported in #2770.

Reply:
I think that makes sense

Reply:
I see also though this was discussed in the hack md notes for May as maybe not ideal? ( or that's saying that raising this warning to indicate about a future behaviour change is a bad idea?)

Reply:
I am not sure exactly what was discussed about the warnings but I don't think it was this one. We mainly talked about ways of keeping the changed behaviour.. One thing we mentioned was that a warning that cannot be silenced is not optimal but this is not that case as you can easily pass `named_geometry.array` instead.